walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,11 +27,11 @@
 
   // Returns available times for meeting requesters based on the events and its attendees' schedules.
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    Set<String> optionalAttendees = new HashSet(request.getOptionalAttendees());
+    Set<String> mandatoryAttendees = new HashSet(request.getAttendees());
     Set<String> allAttendees = getAllAttendees(events);
     List<TimeRange> availableTimeRanges = new ArrayList();
 
-    List<Event> eventsListOfMandatory = createSortedEventsListOfMandatory(events, optionalAttendees);
+    List<Event> eventsListOfMandatory = createSortedEventsListOfMandatory(events, mandatoryAttendees);
     List<Event> eventsListOfMandatoryAndOptional = createSortedEventsListOfMandatoryAndOptional(events);
```

Comment:
I'd imagine there are events that don't have any of the mandatoree attendees nor the optional attendees.

Ie;  event has attendees A, B, C
      required attendee is D

we shouldn't add that event to the list of mandatory and optional.

Reply:
Hello Tay, lines 41-44 check whether the set of requesting attendees overlap with the attendees of the meeting being scheduled. Would moving this code before creating the sorted events list of mandatory and mandatory+optional solve this? Thank you!

Reply:
I don't think so; the problem is if you had two events;

Event 1 - attendees A,B,C
Event 2 - D

lines 41-44 would see an overlap, but later you would exclude any times from event 1